AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -212,16 +212,37 @@ - (BOOL)validateLogName:(NSString *)name {
 }
 
 - (BOOL)validateLogData:(MSCSData *)data {
-  NSDictionary<NSString *, NSString *> *properties = data.properties;
+  return [self validateProperties:data.properties];
+}
+
+- (BOOL)validateProperties:(NSDictionary<NSString *, id> *) properties {
+  BOOL valueIsAString;
   for (NSString *key in properties) {
-    if (![key isKindOfClass:[NSString class]] ||
-        ![properties[key] isKindOfClass:[NSString class]]) {
+    BOOL keyIsNSString = [key isKindOfClass:[NSString class]];
+    BOOL valueIsADictionary = [properties[key] isKindOfClass:[NSDictionary class]];
+    valueIsAString = [properties[key] isKindOfClass:[NSString class]];
+    if (!keyIsNSString) {
       MSLogError([MSAppCenter logTag],
-                 @"%@ Properties key and value must be of type NSString.",
+                 @"%@ Property key must be of type NSString.",
                  kMSBaseErrorMsg);
       return NO;
     }
+    else {
+      if(valueIsADictionary) {
+        
+        // Properties can be nested, verify the nested properties here.
+        return [self validateProperties:properties[key]];
```

Comment:
This will stop validating remaining key/value pairs. You should only return `NO` if nested property validation failed. Otherwise, you should continue the loop.

Reply:
Good catch! thx!